src/bin/stratis_dumpmetadata.rs

```diff
@@ -21,7 +21,13 @@ fn run(devpath: &str) -> Result<(), String> {
         .ok_or_else(|| "No Stratis BDA metadata found".to_string())?;
     println!("{:#?}", bda);
 
-    devfile.seek(SeekFrom::Start(0)).unwrap();
+    match devfile.seek(SeekFrom::Start(0)) {
```

Comment:
It makes much more sense to handle this just the same as all the other ```unwrap()```s.

The things you want are:

* a useful error
* that there will be an immediate exit from this function if the seek fails

and all that will be handled by use of ```map_err()``` and the ```?``` operator.

Reply:
...oh, that's why it didn't work.  I had abandoned the entire `map_err` statement for a `match` block, because I was getting "unused Result which must be used".

All that over a single `?` character.

Reply:
Ah. You can imagine that ```?``` character as behaving like a macro. So,

```
<expr>?;
```

is more or less equivalent to:

```
match <expr> {
  Err(err) => return Err(err),
  Ok(thing) => thing,
};
```

there's a bit more to it than that, actually, but that _is_ the basic idea.